Source/Core/VideoCommon/ShaderCache.cpp

```diff
@@ -627,6 +617,22 @@ static GXPipelineUid ApplyDriverBugs(const GXPipelineUid& in)
     ps->ztest = EmulatedZ::Early;
   }
 
+  // We can use framebuffer fetch to emulate logic ops in the fragment shader.
+  // So don't use the worse blend approximation.
+  if (blend.logicopenable && !g_ActiveConfig.backend_info.bSupportsLogicOp &&
+      !g_ActiveConfig.backend_info.bSupportsFramebufferFetch)
+  {
+    if (!blend.LogicOpApproximationIsExact())
+      WARN_LOG_FMT(VIDEO,
+                   "Approximating logic op with blending, this will produce incorrect rendering.");
+    if (blend.LogicOpApproximationWantsShaderHelp())
+    {
+      ps->emulate_logic_op_with_blend = true;
+      ps->logic_op_mode = static_cast<u32>(blend.logicmode.Value());
+    }
+    blend.ApproximateLogicOpWithBlending();
```

Comment:
Sorry, was trying to tidy up my comments as I make sense of this.  Please see the above comment.

Reply:
You'd still need the `LogicOpApproximationIsExact` check on the warning, because `Invert` and `NoOp` are exact but need blending.

Reply:
Yeah, I guess I'm just saying it's a little hard to read/follow as written.

Something like:

````
if (blend.IsLogicOpsEmulatedWithBlending())
{
  blend.EmulateLogicOpsWithBlending();
  ps->emulate_logic_op_with_blend = true; 
  ps->logic_op_mode = static_cast<u32>(blend.logicmode.Value());
}
else if (blend.IsLogicOpsApproximatedWithBlending())
{
  WARN_LOG_FMT(VIDEO, "Approximating logic op with blending, this will produce incorrect rendering.");
  ps->emulate_logic_op_with_blend = false;
  blend.ApproximateLogicOpWithBlending();
}
```

would be a bit clearer in my mind.

Still, I know that's fairly pedantic so feel free to take it or leave it.